src/MongoDB/Driver/Manager.cpp

```diff
@@ -476,6 +478,12 @@ void HHVM_METHOD(MongoDBDriverManager, __construct, const String &dsn, const Arr
 	hippo_mongo_driver_manager_apply_rp(uri, options);
 	hippo_mongo_driver_manager_apply_wc(uri, options);
 
+	if (options.exists(s_MongoDBDriverManager_appname)) {
+		if (!mongoc_uri_set_appname(uri, options[s_MongoDBDriverManager_appname].toString().c_str())) {
+			throw MongoDriver::Utils::throwInvalidArgumentException("Failed to set appname '" + options[s_MongoDBDriverManager_appname].toString() + "'");
```

Comment:
Thoughts on `"Invalid appname value: '%s'"`? It's roughly consistent with our `"Unsupported readPreference value: %s"` error.